common/flatpak-dir.c

```diff
@@ -309,7 +309,18 @@ flatpak_remote_state_ensure_metadata (FlatpakRemoteState *self,
 
       /* If the collection ID is NULL the metadata comes from the summary */
       if (self->metadata_fetch_error != NULL)
-        error_msg = g_strdup (self->metadata_fetch_error->message);
+        {
+          /* Fetch error can be IO_ERROR_CANCELLED in instances like quick searching of apps in software clients.
+           * Hence, avoid converting them to FLATPAK_ERROR from IO_ERROR domain so the clients can handle the
+           * cancellation appropriately. */
+          if (g_error_matches (self->metadata_fetch_error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
+            {
+              g_propagate_error (error, g_steal_pointer (&self->metadata_fetch_error));
+              return flatpak_fail (error, _("Unable to load metadata from remote %s"), self->remote_name);
```

Comment:
I don't think this can be correct. `flatpak_fail` is an alias for `glnx_throw` which eventually calls `g_propagate_error (error, ...)`, but since you have called `g_propagate_error` on the line above, this is a double-assignment to `*error` and a warning will be issued.

I think you want something more like:

```c
g_propagate_prefixed_error (error, g_steal_pointer (&self->metadata_fetch_error), _("Unable to load metadata from remote %s: "), self->remote_name);
```

Reply:
Nice catch! Thanks. Addressed in the latest iteration.